### ISSUE TITLE
Implement and use virtual batch norm instead of regular batch norm.

### ISSUE THREAD

**Issue by dansuh17** [deNsuh/segan-pytorch#1]

**Comment by dansuh17** [comment]
https://discuss.pytorch.org/t/parameter-grad-of-conv-weight-is-none-after-virtual-batch-normalization/9036/6
